### DISCUSSION TITLE
Error while building the next.js project on github

### DISCUSSION THREAD

**Discussion by DurgaiVS** [community/community#28637] [Actions]
I tried building and deploying my next.js project on github, but there seems to be an error in installing dependencies to that.
I couldn't figureout what causes this error. Below is the debugged details of the error. Hope, will get some help here.

The project link is, 
 https://github.com/DurgaiVS/crowdCoinWeb3App

:

Run npm ci
npm ERR! code ERESOLVE
npm ERR! ERESOLVE could not resolve
npm ERR! 
npm ERR! While resolving: next-routes@[1](https://github.com/DurgaiVS/crowdCoinWeb3App/runs/7709484097?check_suite_focus=true#step:7:1).[4](https://github.com/DurgaiVS/crowdCoinWeb3App/runs/7709484097?check_suite_focus=true#step:7:5).2
npm ERR! Found: react@18.2.0
npm ERR! node_modules/react
npm ERR!   react@"^18.2.0" from the root project
npm ERR!   peer react@"^16.8.0 || ^17 || ^18" from @fluentui/react-component-event-listener@0.63.1
npm ERR!   node_modules/@fluentui/react-component-event-listener
npm ERR!     @fluentui/react-component-event-listener@"~0.63.0" from semantic-ui-react@2.1.3
npm ERR!     node_modules/semantic-ui-react
npm ERR!       semantic-ui-react@"^2.1.3" from the root project
npm ERR!   8 more (@fluentui/react-component-ref, ...)
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer react@"^1[5](https://github.com/DurgaiVS/crowdCoinWeb3App/runs/7709484097?check_suite_focus=true#step:7:6).4.2 || ^1[6](https://github.com/DurgaiVS/crowdCoinWeb3App/runs/7709484097?check_suite_focus=true#step:7:7)" from next-routes@1.4.2
npm ERR! node_modules/next-routes
npm ERR!   next-routes@"^1.4.2" from the root project
npm ERR! 
npm ERR! Conflicting peer dependency: react@16.14.0
npm ERR! node_modules/react
npm ERR!   peer react@"^15.4.2 || ^16" from next-routes@1.4.2
npm ERR!   node_modules/next-routes
npm ERR!     next-routes@"^1.4.2" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! See /home/runner/.npm/eresolve-report.txt for a full report.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/runner/.npm/_logs/2022-08-0[7](https://github.com/DurgaiVS/crowdCoinWeb3App/runs/7709484097?check_suite_focus=true#step:7:8)T03_43_15_[8](https://github.com/DurgaiVS/crowdCoinWeb3App/runs/7709484097?check_suite_focus=true#step:7:9)5[9](https://github.com/DurgaiVS/crowdCoinWeb3App/runs/7709484097?check_suite_focus=true#step:7:10)Z-debug-0.log

**Comment by supervoid0** [comment]
This is a conflict between dependencies. The version mismatch between direct and indirect dependency. To automatically solve the issue without manually resolving version mismatch, follow these these steps - 

1. Go to the project folder and remove **node_modules** and **package-lock.json**.
2. Run any one of these commands - `npm install  --legacy-peer-deps` Or `npm install  --force`

**Reply by DurgaiVS** [reply]
Tried both commands after removing the **node_modules** and **package-lock.json**, still getting the same error. :(

**Reply by supervoid0** [reply]
![npm](https://user-images.githubusercontent.com/58064770/183831634-49cd5c7a-aa9b-4216-9ebf-e4dc1b2fcdd7.png)

I was able to do it.  Remove node_modules and package.lock again and try `npm install --force` again
@DurgaiVS

**Reply by DurgaiVS** [reply]
Sry, my fault. Now working.... Thank you very much @supervoid0 , means a lot.  : )
